cmd/sops/rotate.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/codes"
+	"go.mozilla.org/sops/cmd/sops/common"
+	"go.mozilla.org/sops/keys"
+	"go.mozilla.org/sops/keyservice"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+type RotateOpts struct {
+	Cipher           sops.DataKeyCipher
+	InputStore       sops.Store
+	OutputStore      sops.Store
+	InputPath        string
+	IgnoreMAC        bool
+	AddMasterKeys    []keys.MasterKey
+	RemoveMasterKeys []keys.MasterKey
+	KeyServices      []keyservice.KeyServiceClient
+}
+
+func Rotate(opts RotateOpts) ([]byte, error) {
+	tree, err := common.LoadEncryptedFile(opts.InputStore, opts.InputPath)
+	if err != nil {
+		return nil, err
+	}
+
+	dataKey, err := common.DecryptTree(common.DecryptTreeOpts{
+		Stash: make(map[string][]interface{}), Cipher: opts.Cipher, IgnoreMac: opts.IgnoreMAC, Tree: tree,
+		KeyServices: opts.KeyServices,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Add new master keys
+	for _, key := range opts.AddMasterKeys {
+		tree.Metadata.KeyGroups[0] = append(tree.Metadata.KeyGroups[0], key)
+	}
+	// Remove master keys
+	for _, rmKey := range opts.RemoveMasterKeys {
+		for i := range tree.Metadata.KeyGroups {
+			for j, groupKey := range tree.Metadata.KeyGroups[0] {
```

Comment:
this doesn't appear to be fixed per above comment, need to iterate over tree.Metadata.Keygroups[i] not [0]

Reply:
Fixed, nice catch